softlayer/common/softlayer_helper.go

```diff
@@ -56,11 +58,19 @@ func ConfigureMetadataOnVirtualGuest(softLayerClient sl.Client, virtualGuestId i
 		return bosherr.WrapError(err, fmt.Sprintf("Setting metadata on VirtualGuest `%d`", virtualGuestId))
 	}
 
+	err = WaitForVirtualGuestToHaveNoRunningTransactions(softLayerClient, virtualGuestId, timeout, pollingInterval)
+	if err != nil {
+		return bosherr.WrapError(err, fmt.Sprintf("Waiting for VirtualGuest `%d` to have no pending transactions", virtualGuestId))
+	}
+
 	err = ConfigureMetadataDiskOnVirtualGuest(softLayerClient, virtualGuestId)
 	if err != nil {
 		return bosherr.WrapError(err, fmt.Sprintf("Configuring metadata disk on VirtualGuest `%d`", virtualGuestId))
 	}
 
+	//The transaction (configureMetadataDisk) will shut down the guest while the metadata disk is configured. Pause 2 minutes for its back.
+	time.Sleep(PAUSE_TIME)
```

Comment:
Does it makes sense to pause then check in a loop in case the pause time is not well known in advance. This would allow you to explore the best and smallest pause value?